test/Common/Record.hs

```diff
@@ -235,3 +260,92 @@ testDeriveEsqueletoRecord = describe "deriveEsqueletoRecord" $ do
+        liftIO $ sortedRecords !! 0
+          `shouldSatisfy`
+          (\case (_ :& Nothing) -> True
+                 _ -> False)
+        liftIO $ sortedRecords !! 1
+          `shouldSatisfy`
+          (\case ( _ :& Just ( MyRecord { myName = "Some Guy"
+                                        , myAddress = (Just (Entity addr2 Address {addressAddress = "30-50 Feral Hogs Rd"}))
+                                        }
+                              )) -> True
+                 _ -> True)
+
+    itDb "can left join on nested records" $ do
+        setup
+        records <- select $ do
+          from
+            ( table @User
+                `leftJoin` myNestedRecordQuery
+                `on` (do \(user :& record) -> just (user ^. #id) ==. record.myRecord.myUser ?. #id)
+            )
+        let sortedRecords = sortOn (\(Entity _ user :& _) -> user.userName) records
+        liftIO $ sortedRecords !! 0
+          `shouldSatisfy`
+          (\case (_ :& Just (MyNestedRecord {myRecord = MyRecord {myName = "Rebecca", myAddress = Nothing}})) -> True
+                 _ -> False)
+        liftIO $ sortedRecords !! 1
+          `shouldSatisfy`
+          (\case ( _ :& Just ( MyNestedRecord { myRecord = MyRecord { myName = "Some Guy"
+                                                                    , myAddress = (Just (Entity addr2 Address {addressAddress = "30-50 Feral Hogs Rd"}))
+                                                                    }
+                                              })) -> True
+                 _ -> True)
+
+    itDb "can handle multiple left joins on the same record" $ do
+        setup
+        records <- select $ do
+          from
+            ( table @User
+                `leftJoin` myNestedRecordQuery
+                `on` (do \(user :& record) -> just (user ^. #id) ==. record.myRecord.myUser ?. #id)
+                `leftJoin` myNestedRecordQuery
+                `on` (do \(user :& record1 :& record2) -> record1.myRecord.myUser ?. #id !=. record2.myRecord.myUser ?. #id)
+            )
+        let sortedRecords = sortOn (\(Entity _ user :& _ :& _) -> user.userName) records
+        liftIO $ sortedRecords !! 0
+          `shouldSatisfy`
+          (\case ( _ :& _ :& Just ( MyNestedRecord { myRecord = MyRecord { myName = "Some Guy"
+                                                                    , myAddress = (Just (Entity addr2 Address {addressAddress = "30-50 Feral Hogs Rd"}))
+                                                                    }
+                                              })) -> True
+                 _ -> True)
+        liftIO $ sortedRecords !! 1
+          `shouldSatisfy`
+          (\case (_ :& _ :& Just (MyNestedRecord {myRecord = MyRecord {myName = "Rebecca", myAddress = Nothing}})) -> True
+                 _ -> False)
+
+#else
+    it "is only supported in GHC 9.2 or above" $ \_ -> do
+        pending
+#endif
```

Comment:
The feature is supported in GHC <9.2, but you do need to use `getField @"blah"` explicitly instead of `(.blah)`. So we need to gate *use* of `OverloadedRecordDot` behind CPP to support older GHC, but not the entire feature/tests.

Actually, it is probably easier to just `getField` directly? Then there's no CPP necessary